### PR TITLE
Separate data model errors from syntax errors

### PR DESCRIPTION
**Pull request by catamorphism**
Split out data model errors into their own file and label tests according to the error name from the spec.

No new test cases are added/removed.

### CODE REVIEW

**Comment by catamorphism** [comment]
When integrating these tests into my implementation, I was surprised to find that not all the tests in `syntax-errors.json` were actually syntax errors, and I thought it was also nice to label the tests according to which data model error they should trigger, since it's not immediately obvious.

File: test/README.md [hunk 1]
```diff
@@ -3,7 +3,13 @@ and are here relicensed by their original author (Eemeli Aro) under the Unicode
...
 These test files are intended to be useful for testing multiple different message processors in different ways:
 
-- `syntax-errors.json` — An array of strings that should produce a Syntax Error or a Data Model Error when parsed.
+- `syntax-errors.json` — An array of strings that should produce a Syntax Error when parsed.
+
+- `data-model-errors.json` - An array of pairs of error names and arrays of strings:
+    - `error: string` (required) - The name of the error; names are defined in
+        ["MessageFormat 2.0 Errors"](https://github.com/unicode-org/message-format-wg/blob/main/spec/errors.md)
+        in the spec.
+    - `tests: string[]` (required): An array of strings that should produce `error` when processed.
```
Comment (eemeli) [inline test/README.md:12]: Why not an object `{ [error: string]: string[] }` as in `test-function.json`?

**Review by aphillips — APPROVED** [review]
... assuming @eemeli's comment gets addressed

File: test/README.md (hunk 1 above)
Comment (catamorphism) [reply test/README.md]: No good reason -- changed it in [e3139f1](https://github.com/unicode-org/message-format-wg/pull/708/commits/e3139f1a162b8c3500b005fb2fd2eb9de2b0029d) to the format you suggested.

File: test/syntax-errors.json [hunk 2]
```diff
@@ -41,33 +39,15 @@
...
-  ".local $foo = {$bar :func} .local $bar = {$baz} {{_}}",
-  ".local $foo = {42 :func opt=$foo} {{_}}",
-  ".local $foo = {42 :func opt=$bar} .local $bar = {42} {{_}}",
   ".match {#foo} * {{foo}}",
   ".match {} * {{foo}}",
   ".match {|foo| :x} {|bar| :x} ** {{foo}}",
   ".match * {{foo}}",
   ".match {|x| :x} * foo",
   ".match {|x| :x} * {{foo}} extra",
   ".match |x| * {{foo}}",
```
Comment (eemeli) [inline test/syntax-errors.json:52]:
```suggestion
  ".match |x| * {{foo}}"
```

File: test/README.md [hunk 3]
```diff
@@ -3,7 +3,14 @@ and are here relicensed by their original author (Eemeli Aro) under the Unicode
...
 
-- `syntax-errors.json` — An array of strings that should produce a Syntax Error or a Data Model Error when parsed.
+- `syntax-errors.json` — An array of strings that should produce a Syntax Error when parsed.
+
+- `data-model-errors.json` - An object with string keys and arrays of strings as values,
+     where each key is the name of an error and its value is array of strings that
+     should produce `error` when processed.
+     Error names are defined in
+        ["MessageFormat 2.0 Errors"](https://github.com/unicode-org/message-format-wg/blob/main/spec/errors.md)
+        in the spec.
```
Comment (eemeli) [inline test/README.md:13]: With a relative path this'll keep pointing to the same branch.

```suggestion
     Error names are defined in ["MessageFormat 2.0 Errors"](../spec/errors.md) in the spec.
```

**Review by eemeli — APPROVED** [review]
Looks good, pending two small inline fixes.

**Pull request merged**